feat(post-scan action): Create terraform version for the aws-slack-plugin (#102)
trendmicro/cloudone-filestorage-plugins: post-scan-actions/aws-python-slack-notification/terraform/README.md

# Post Scan Action - Slack Notification

> After a scan occurs and a malicious file is detected, this example Lambda function sends out a notification to Slack using Amazon Simple Notification Service.

Reviewer: Do we need the term "example"? "Sample" would be better, and removing it altogether would be better.
```suggestion
After a scan occurs and a malicious file is detected, this Lambda function sends out a notification to Slack using Amazon Simple Notification Service.
```

## Prerequisites

1. **Install supporting tools**
    Do one of the following:
    - Install the AWS command line interface (CLI). See [Installing the AWS CLI](https://docs.aws.amazon.com/cli/latest/userguide/cli-chap-install.html) for details.

>     - Install Terraform CLI. See [Install Terraform CLI](https://learn.hashicorp.com/tutorials/terraform/install-cli#install-terraform) for download information.

Reviewer: ```suggestion
    - Install the Terraform CLI. See [Install Terraform CLI](https://learn.hashicorp.com/tutorials/terraform/install-cli#install-terraform) for download information.
```

2. **Configure Slack Webhook App**
    - Create a Slack Channel to receive the notification
    - Go to App Directory > Search `Incoming WebHooks`.
    - Click on `Incoming WebHooks`, then click "Add to Slack"
    - Choose the Channel to receive the notification
    - Copy Webhook URL
    - Enter the Description of your WebHook.
    - Enter the Name of the Slack WebHook, by default it will use `incoming-webhook`; if you prefer, you can customize the name.

>     - If you want any custom icon to add that in Customize Icon section.

Reviewer: ```suggestion
    -You can add a custom icon in Customize Icon section.
```

    - Click "Save Setting"
    
    [Additional information](https://slack.com/help/articles/115005265063-Incoming-webhooks-for-Slack)

> 3. Make sure that you have terraform and AWS CLI installed and configured, then go to terraform folder and change appropriately the values at the variables in the `variables.tf` file.

Reviewer 2: We can tell the user to create a tfvars file to provide the values of input variables instead of directly changing the `variables.tf`. How about providing a tfvars.example file for the user to reference? 

https://www.terraform.io/language/values/variables#variable-definitions-tfvars-files

Author: We definitely could, but I don't see much need since is a very simple plugin and has very few variables (only 5). It is also possible to add during the plan, for example:

```
terraform plan -var region="us-west-1" -var SlackChannel="dev" ...
```

Any specific benefit of using the `tfvars` in that case that I may be missing?

Author: Done!

> 3. Make sure that you have terraform and AWS CLI installed and configured, then go to terraform folder and change appropriately the values at the variables in the `variables.tf` file.

Reviewer: ```suggestion
3. Ensure that you have the terraform and AWS CLIs installed and configured, then go to terraform folder and change appropriately the values at the variables in the `variables.tf` file.
```

>     Then you can execute the following:

Reviewer: ```suggestion
    Run the following script:
```

    ```terraform
        terraform init
        terraform plan
        terraform apply
    ```

## Clean-up

> In case you want to revert/destroy the resources created, execute the following:

Reviewer: ```suggestion
To revert or destroy the resources created, run the following script:
```
